### PR TITLE
Edge 109. Updated the migration procedure.

### PR DESCRIPTION
**Pull request by sharath-jk**

### CODE REVIEW

**Review by chandrababumerla — APPROVED** [review]

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+title: Sample scripts to automate the migration
+layout: redirect
+---
+{{< company-sag >}} provides the `backup.sh` and `restore.sh` scripts for your reference. You may customize these scripts for your requirements and automate the migration process. These scripts are available for reference at:
+
+- [backup.sh](/files/edge/backup.sh)
+- [restore.sh](/files/edge/restore.sh) - you must run the restore.sh script as a **root** user.
+- [restore_analytics.sh](/files/edge/restore_analytics.sh) - restores the Streaming Analytics application.
+
+>**IMPORTANT:** {{< company-sag >}} does not officially support these scripts. These scripts are only for your reference.
```
Comment (MWindel) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:12]: @sharath-jk see comments above

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+### Using the scripts
+
+#### In 10.7 appliance
+
+>**Important:** Before you back up the data, ensure that there is sufficient disk space to save the backup in your Edge 10.7 appliance. The MongoDB backup requires the same amount of space as the database. For example, if the size of the database is 100 GB, the MongoDB backup also requires 100 GB of disk space. You would need additional 100 GB of disk space to save the MongoDB backup in your Edge 10.7 appliance.
```
Comment (MWindel) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:18]: @sharath-jk see comments above

Comment (MWindel) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:18]:
```suggestion
>**Important:** Before the data back up, ensure that there is sufficient disk space to save the backup in your Edge 10.7 appliance. The MongoDB backup requires the same amount of space as the database. For example, if the size of the database is 100 GB, the MongoDB backup also requires 100 GB of disk space. You would need additional 100 GB of disk space to save the MongoDB backup in your Edge 10.7 appliance.
```

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+1. Run the following command to stop monitoring all the processes:
+
+    ```shell
+    monit unmonitor all
+    ```
+2. Stop the Karaf process using the command:
```
Comment (MWindel) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:25]:
```suggestion
2. Stop the Karaf process by using the following command:
```

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    ```shell
+    sudo service cumulocity-core-karaf stop
+    ```
+2. Copy the `backup.sh` script to your Edge 10.7 appliance.
+
+3. Run the `backup.sh` as a **root** user.
```
Comment (MWindel) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:32]:
```suggestion
3. Run the `backup.sh` as a root user.
```

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+   You can also run the script with the parameters:
+   - OUTPUT_DIRECTORY: (optional) path to save the backup archive on the same file system.
+   - ARCHIVE_PATH: (optional) path to save the backup archive on an external file system.
+
+    >**Info:** If you do not specify any parameter, the backup archive is saved at */tmp* directory. The */tmp* directory is located on the installation disk. If the installation disk has no space, the system could become unstable.
```
Comment (MWindel) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:38]: @sharath-jk please use our admonitions for this info box

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    ```shell
+    ./backup.sh /home/admin/
+    ```
+   The script creates a ZIP archive file with the migration data in the OUTPUT_DIRECTORY.
+
+4. Move the ZIP archive with the migration data to your Edge 10.9 appliance.
+
+#### In 10.9 appliance
+
+>**Important:** Before copying the backup, ensure that there is sufficient disk space in your Edge 10.9 appliance. For example, in the Edge 10.9 appliance, if the size of the data disk is 75 GB and the size of the MongoDB backup is 100 GB, you must expand the size of the data disk to additional 100 GB before copying the MongoDB backup. For more information about disk size expansion, see [Expanding the disk size](/edge/configuration/#expanding-the-disk-size).
```
Comment (MWindel) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:50]: @sharath-jk see comments above

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+1. Log in as **root** user.
```
Comment (MWindel) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:52]:
```suggestion
1. Log in as root user.
```

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    ./restore_analytics.sh USERNAME PASSWORD /tmp/streaming-analytics-app.zip
+    
+    Here:
+    - USERNAME and PASSWORD refers to the {{< management-tenant >}} user credentials.
+    ```
+Running the `restore_analytics.sh` script completes the migration process.
+
+Next, you must configure the Edge 10.9 appliance. For example, if you had enabled microservices and configured NTP in the Edge 10.7 appliance, you must enable microservices and configure NTP in the Edge 10.9 appliance.
+
+>**Important:** To enable the microservice hosting feature, the {{< management-tenant >}} user must have the "Tenant Manager" role. Use the 10.7 {{< management-tenant >}} admin credentials. By default, the credentials are sysadmin/sysadmin-pass.
```
Comment (MWindel) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:78]: @sharath-jk see comments above.

File: content/edge/migration-bundle/before-you-begin.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
+---
+weight: 10
+title: Before you begin
+layout: redirect
+---
+- Import the Edge 10.9 appliance. See, [Configuring the Edge infrastructure](/edge/setting-up-edge/).
+- Configure the network and complete the installation procedure on Edge 10.9 appliance. See, [Installing {{< product-c8y-iot >}} Edge](/edge/installation/)
+
+>**Important:** You can have both the Edge 10.7 and 10.9 appliances on the same host machine. Ensure that the IP address of the Edge 10.9 appliance is different from Edge 10.7 appliance.
```
[the remark this answers is not in the document]
Comment (rssag) [reply content/edge/migration-bundle/before-you-begin.md]: this is for 10.9.0 - our new admonitions were not introduced yet in 10.9.0, so no change required :)

File: content/edge/migration-bundle/before-you-begin.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+For information about upgrading from an earlier version to {{< product-c8y-iot >}} Edge 10.7, see:
+
+- [https://{{< domain-c8y >}}/guides/10.7.0/edge/operation/#upgrade_esxi](https://{{< domain-c8y >}}/guides/10.7.0/edge/operation/#upgrade_esxi)
+- [https://{{< domain-c8y >}}/guides/10.7.0/edge/operation/#upgrade_vmware_workstation](https://{{< domain-c8y >}}/guides/10.7.0/edge/operation/#upgrade_vmware_workstation)
+- [https://{{< domain-c8y >}}/guides/10.7.0/edge/operation/#upgrade_hyper_v](https://{{< domain-c8y >}}/guides/10.7.0/edge/operation/#upgrade_hyper_v)
```
Comment (rssag) [inline content/edge/migration-bundle/before-you-begin.md:15]:
```suggestion
- [Operating Cumulocity IoT Edge > Upgrading on VMware ESX](https://{{< domain-c8y >}}/guides/10.7.0/edge/operation/#upgrade_esxi),
- [Operating Cumulocity IoT Edge > Upgrading on VMware Workstation Player](https://{{< domain-c8y >}}/guides/10.7.0/edge/operation/#upgrade_vmware_workstation),
- [Operating Cumulocity IoT Edge > Upgrading on Hyper-V](https://{{< domain-c8y >}}/guides/10.7.0/edge/operation/#upgrade_hyper_v)

in the *10.7.0 Cumulocity IoT Edge guide*.
```

File: content/edge/migration-bundle/before-you-begin.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+To migrate from Edge 10.7 to 10.9, you must:
+- first back up the data on Edge 10.7
+- move the backup to Edge 10.9
+- restore the data on Edge 10.9
```
Comment (rssag) [inline content/edge/migration-bundle/before-you-begin.md:20]:
```suggestion
To migrate from Edge 10.7 to 10.9:

- back up the data on Edge 10.7,
- move the backup to Edge 10.9,
- restore the data on Edge 10.9.
```

File: content/edge/migration-bundle/creating-backup-on-107.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
+---
+weight: 20
+title: Creating a backup on Edge 10.7
+layout: redirect
+---
+
+In your Edge 10.7 setup, you must back up the data for each tenant and docker collection, and note down the device ID.
+
+>**Important:** Before you back up the data, ensure that there is sufficient disk space to save the backup in your Edge 10.7 appliance. The MongoDB backup requires the same amount of space as the database. For example, if the size of the database is 100 GB, the MongoDB backup also requires 100 GB of disk space. You would need additional 100 GB of disk space to save the MongoDB backup in your Edge 10.7 appliance.
```
[the remark this answers is not in the document]
Comment (rssag) [reply content/edge/migration-bundle/creating-backup-on-107.md:9]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/restoring-the-data-on-edge-109.md [hunk 13]
```diff
@@ -0,0 +1,164 @@
+---
+weight: 30
+title: Restoring the data on Edge 10.9
+layout: redirect
+---
+To restore the data, you must first copy the MongoDB backup from Edge 10.7 appliance to your Edge 10.9 appliance.
+
+>**Important:** Before copying the backup, ensure that there is sufficient disk space in your Edge 10.9 appliance. For example, in the Edge 10.9 appliance, if the size of the data disk is 75 GB and the size of the MongoDB backup is 100 GB, you must expand the size of the data disk to additional 100 GB before copying the MongoDB backup. For more information about disk size expansion, see [Expanding the disk size](/edge/configuration/#expanding-the-disk-size).
```
[the remark this answers is not in the document]
Comment (rssag) [reply content/edge/migration-bundle/restoring-the-data-on-edge-109.md:8]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/restoring-the-data-on-edge-109.md [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+
+    If your Edge appliance is not connected to the internet, ensure that the RPM package is available locally and run the command:
+
+    ```shell
+    rpm -ivh zip-3.0-11.el7.x86_64.rpm
+    ```
+
+5. Prepare the applications for deployment using the commands:
+
+   >**Important:** Do not include the *streaming-analytics-app.zip* file in the ZIP package.
```
[the remark this answers is not in the document]
Comment (rssag) [reply content/edge/migration-bundle/restoring-the-data-on-edge-109.md:82]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/restoring-the-data-on-edge-109.md [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+
+12. Restore the Streaming Analytics application.
+
+	- Log in to the {{< management-tenant >}} using the 10.7 {{< management-tenant >}} admin credentials. By default, the credentials are sysadmin/sysadmin-pass.
+
+	- Upload the *streaming-analytics-app.zip* file as a web application.
+
+	- Subscribe the Streaming Analytics application to the edge tenant.
+
+      >**Important:** To subscribe the application, the {{< management-tenant >}} user must have the "Tenant Manager" role.
```
[the remark this answers is not in the document]
Comment (rssag) [reply content/edge/migration-bundle/restoring-the-data-on-edge-109.md:150]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/restoring-the-data-on-edge-109.md [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+
+	- Delete the Apama Analytics Builder and Apama EPL Apps applications.
+
+	- Log in to the edge tenant and verify the Streaming Analytics application.
+
+Restoring the Streaming Analytics application completes the migration procedure. Note that the tenants from Edge 10.9 installation are removed after the migration is successful. You will now be able to log in using the Edge 10.7 user credentials.
+
+Next, you must configure the Edge 10.9 appliance. For example, if you had enabled microservices and configured NTP in the Edge 10.7 appliance, you must enable microservices and configure NTP in the Edge 10.9 appliance.
+
+>**Important:** To enable the microservice hosting feature, the {{< management-tenant >}} user must have the "Tenant Manager" role. Use the 10.7 {{< management-tenant >}} admin credentials. By default, the credentials are sysadmin/sysadmin-pass.
```
[the remark this answers is not in the document]
Comment (rssag) [reply content/edge/migration-bundle/restoring-the-data-on-edge-109.md:160]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md (hunk 1 above)
Comment (rssag) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:12]:
```suggestion
>**Important:** {{< company-sag >}} does not officially support these scripts. These scripts are only for your reference.
```

Comment (rssag) [reply content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:12]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md (hunk 2 above)
Comment (rssag) [reply content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:18]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md (hunk 5 above)
Comment (rssag) [reply content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:38]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md (hunk 6 above)
Comment (rssag) [reply content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:50]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md (hunk 8 above)
Comment (rssag) [reply content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:78]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/before-you-begin.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (sharath-jk) [reply content/edge/migration-bundle/before-you-begin.md]: @MWindel - the important and info boxes shortcodes were introduced only after 10.11 release. These changes are for 10.9 release.

File: content/edge/migration-bundle/restoring-the-data-on-edge-109.md [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+     - APP_NAME refers to the name of the application. For example, devicemanagement.zip
+
+    For example:
+    mongofiles -d management --prefix cmdata get  112 -l /tmp/apps/cockpit.zip
+    mongofiles -d management --prefix cmdata get  113 -l /tmp/apps/devicemanagement.zip
+    mongofiles -d management --prefix cmdata get  114 -l /tmp/apps/administration.zip
+    mongofiles -d management --prefix cmdata get  122 -l /tmp/apps/streaming-analytics-app.zip
+    ```
+
+    >**Important:** Create a backup of the streaming-analytics-app.zip file separately.
```
[the remark this answers is not in the document]
Comment (rssag) [reply content/edge/migration-bundle/restoring-the-data-on-edge-109.md:66]: see above, new admons not yet available in 10.9.0

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md (hunk 6 above)
Comment (sharath-jk) [reply content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:50]: @MWindel - the important and info boxes shortcodes were introduced only after 10.11 release. These changes are for 10.9 release.

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md (hunk 8 above)
Comment (sharath-jk) [reply content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:78]: @MWindel - the important and info boxes shortcodes were introduced only after 10.11 release. These changes are for 10.9 release.

**Review by rssag — APPROVED** [review]

File: content/edge/migration-bundle/creating-backup-on-107.md [hunk 18]
```diff
@@ -0,0 +1,30 @@
+---
+weight: 20
+title: Creating a backup on Edge 10.7
+layout: redirect
+---
+
+In your Edge 10.7 setup, back up the data for each tenant and docker collection, and note down the device ID.
+
+>**Important:** Before the data back up, ensure that there is sufficient disk space to save the back up in your Edge 10.7 appliance. The MongoDB backup requires the same amount of space as the database. For example, if the size of the database is 100 GB, the MongoDB backup also requires 100 GB of disk space. You would need additional 100 GB of disk space to save the MongoDB backup in your Edge 10.7 appliance.
```
Comment (rssag) [inline content/edge/migration-bundle/creating-backup-on-107.md:9]:
```suggestion
>**Important:** Before the data back up, ensure that there is sufficient disk space to save the back up in your Edge 10.7 appliance. The MongoDB backup requires the same amount of space as the database. For example, if the size of the database is 100 GB, the MongoDB backup also requires 100 GB of disk space. You would need an additional 100 GB of disk space to save the MongoDB backup in your Edge 10.7 appliance.  
```

File: content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+    ```
+   The script creates a ZIP archive file with the migration data in the OUTPUT_DIRECTORY.
+
+4. Move the ZIP archive with the migration data to your Edge 10.9 appliance.
+
+#### In 10.9 appliance
+
+>**Important:** Before copying the backup, ensure that there is sufficient disk space in your Edge 10.9 appliance. For example, in the Edge 10.9 appliance, if the size of the data disk is 75 GB and the size of the MongoDB backup is 100 GB, you must expand the size of the data disk to additional 100 GB before copying the MongoDB backup. For more information about disk size expansion, see [Expanding the disk size](/edge/configuration/#expanding-the-disk-size).
+
+1. Log in as root user.
```
Comment (rssag) [inline content/edge/migration-bundle/sample-scripts-to-automate-the-migration.md:52]:
```suggestion
1. Log in as a root user.
```

**Pull request merged**